LanguageFeatures/Inline-classes/dynamic_semantics_inline_class_A04_t01.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion An inline type V used as an expression (a type literal) evaluates
+/// to the value of the corresponding instantiated representation type used as
+/// an expression.
+///
+/// @description Check that if an inline type `V` is used as an expression then
+/// it evaluates to the value of the corresponding instantiated representation
+/// type
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+inline class V {
+  final int id;
+  V(this.id);
+}
+
+main() {
+  Type t = V;
+  Expect.equals(int, t);
```

Comment:
Very good! It would be useful to have a generic inline class, too, such that the work done in order to find the instantiated representation type isn't a no-op.

Perhaps something like this:

```dart
inline class V<X> {
  final X Function(int) f;
  V(this.f);
}

typedef typeof<X> = X;

void main() {
  Expect.equals(V<bool>, typeof<bool Function(int)>);
}
```